lib/src/git.rs

```diff
@@ -1424,6 +1430,86 @@ fn build_index_from_merged_tree(
     Ok(index)
 }
 
+pub fn update_intent_to_add(
```

Comment:
nit: for future reference, I think it's better to include when this function should be called.